### ISSUE TITLE
Private IPV6 address is not configurable

### ISSUE THREAD

**Issue by michaeldg** [mail-in-a-box/mailinabox#1738]
Hi,

From what I understand, the feature of public and private IPs is used for miab installations that are behind NAT. 

I have a Linode box and Linode provided me with an extra IPv6 /64 range. This results in my Linode having 3 IPv6 addreses configured. Somehow it seams that I cannot remove the default IP addresses added by Linode.

I have found that it is possible to change the used public IP addresses in the configuration file /etc/mailinabox.conf. The same configuraton file also contains a record 'PRIVATE_IPV6_ADDRESS', but changing this has no effect. The installation mentions the (detected) IP, not the IP from the configuration and changes the configuration file in the process to the (detected, incorrect) IP.

Please change this so we can configure the private IPV6 address, as this is used for outgoing postfix SMTP connections. If this connection is with the wrong IPV6 address the suggested and automatically configured SPF records are incorrect.

Steps to reproduce:
1. Create a cloud VM with two public IPV6 IPs (You can create one with Linode, and request a /64 IPV6 range for mail sending)
2. Install MIAB normally, observe the private IP address that is detected by the installer
3. Change the configuration file to the other IPV6 address
4. Rerun installer
5. Observe the same IPV6 address as in step 2

Thank you!

**Comment by starsoccer** [comment]
+1 having this same issue

**Comment by nugget** [comment]
Also suggest changing the terminology away from “public” and “private” because those terms are meaningless and likely confusing in this context.

**Comment by nilsnh** [comment]
Just sharing how I solved this. After searching the source code I saw that MIAB has env variables for this. 

You can try it out by running `PUBLIC_IPV6=YOUR_IP PRIVATE_IPV6=YOUR_IP mailinabox` it will set env vars for that particular command. After the command is completed you can visit your dashboard and you should see your new IP be used, and no mention of the private ip.

To persist this config I created a file `/etc/profile.d/mailinabox.sh` and added:

```bash
export PUBLIC_IPV6=MY_CUSTOM_IP
export PRIVATE_IPV6=MY_CUSTOM_IP
```

After re-logging into your server you should have these env vars present in your shell. Verify it by calling `echo $PUBLIC_IPV6`.

**Comment by nugget** [comment]
It's interesting that your approach worked @nilsnh because I swear that tecnique didn't work in 2023 when I opened #2262 in an attempt to fix this long-standing frustration with mailinabox.

My current solution (which I've been using for the past three years?) is a local patch to `./mailinabox/setup/mail-postfix.sh` like this:

```
root@mail:~/mailinabox# git diff
diff --git a/setup/mail-postfix.sh b/setup/mail-postfix.sh
index 5a4c7fe..a2c1d58 100755
--- a/setup/mail-postfix.sh
+++ b/setup/mail-postfix.sh
@@ -56,7 +56,7 @@ apt_install postfix postfix-sqlite postfix-pcre postgrey ca-certificates
 tools/editconf.py /etc/postfix/main.cf \
        inet_interfaces=all \
        smtp_bind_address="$PRIVATE_IP" \
-       smtp_bind_address6="$PRIVATE_IPV6" \
+       smtp_bind_address6="$PUBLIC_IPV6" \
        myhostname="$PRIMARY_HOSTNAME"\
        smtpd_banner="\$myhostname ESMTP Hi, I'm a Mail-in-a-Box (Ubuntu/Postfix; see https://mailinabox.email/)" \
        mydestination=localhost
```

This lets me use the normal setup upgrade path most of the time. It only breaks and requires intervention whenever the project has made changes to `mail-postfix.sh` which then cause a merge conflict when I attempt to upgrade. With releases which can't cleanly install, I have to reset that file to stock, run the upgrade, and then re-apply my changes after the upgrade to restore correct operation.

The core issue here is that mailinabox tries to treat IPv6 like the NAT-based hellscape that is IPv4, expecting IPv6 to behave in the same, broken, un-routable way that we're forced to use with IPv4.

There's no such thing (practically speaking) as a "private" IPv6 address. Not like mailinabox thinks, at least.

**Comment by nilsnh** [comment]
> There's no such thing (practically speaking) as a "private" IPv6 address. Not like mailinabox thinks, at least.

Yes, this terminology is unclear. I remember being puzzled by the private vs. public distinction. I guess this is an unfortunate carry-over from the IPv4 logic. Or maybe there some specific IPv6 logic that requires this distinction? 🤷 

With IPv6 it's not uncommon to get multiple IPv6 IPs for a given box: [A SLAAC-based link-local address](https://en.wikipedia.org/wiki/IPv6_address#Stateless_address_autoconfiguration_(SLAAC)) plus one or more globally routable IPv6 addresses.
